src/game/client/tf/tf_hud_weaponselection.cpp

```diff
@@ -437,17 +436,33 @@ void CHudWeaponSelection::ComputeSlotLayout( SlotLayout_t *rSlot, int nActiveSlo
 				{
 					rSlot[i].wide = m_flLargeBoxWide;
 					rSlot[i].tall = m_flLargeBoxTall;
+					nTotalHeight += rSlot[i].tall;
 				}
 				else
 				{
 					rSlot[i].wide = m_flSmallBoxWide;
-					rSlot[i].tall = m_flSmallBoxTall;
+					// only include slot if visible OR (any slot above visible AND any slot below visible)
+					if ( ( iSlotBits >> i ) && ( iSlotBits & ( ( 1 << ( i + 1 ) ) - 1 ) ) )
+					{
+						rSlot[i].tall = m_flSmallBoxTall;
```

Comment:
This line can be replaced with the following in order to reduce the amount of space that empty slots take up in the UI.
```
rSlot[i].tall = ( iSlotBits & ( 1 << i ) ) ? m_flSmallBoxTall : m_flSmallBoxTall / 2;
```

Reply:
Made a version on my fork that adds ConVars to adjust this functionality for mods who wish to add it:
https://github.com/BitlDevelopmentStudios/source-sdk-2013-bds-base/commit/5253d169af66a1b7acce5e388b3f7afcbb387400

Reply:
https://github.com/ValveSoftware/source-sdk-2013/pull/901#issuecomment-2757065649